app/code/core/Mage/Index/Model/Observer.php

```diff
@@ -136,4 +139,35 @@ public function processConfigDataSave(Varien_Event_Observer $observer)
             Mage_Index_Model_Event::TYPE_SAVE
         );
     }
+
+    /**
+     * Clean old index events for indexers in manual mode
+     *
+     * @return void
+     */
+    public function cleanOutdatedEvents()
+    {
+        $manualIndexProcessCollection = Mage::getSingleton('index/indexer')
+            ->getProcessesCollection()
+            ->addFieldToFilter('mode', Mage_Index_Model_Process::MODE_MANUAL);
+
+        $now = new DateTime();
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $dateInterval = new DateInterval('PT' . self::OLD_INDEX_EVENT_THRESHOLD_SECONDS . 'S');
+        $oldEventsThreshold = $now
+            ->sub($dateInterval)
+            ->format('Y-m-d H:i:s');
+
+        /** @var Mage_Index_Model_Process $process */
+        foreach ($manualIndexProcessCollection as $process) {
+            $unprocessedEventsCollection = $process
+                ->getUnprocessedEventsCollection()
+                ->addFieldToFilter('created_at', ['lt' => $oldEventsThreshold])
+                ->load();
+
+            foreach ($unprocessedEventsCollection as $unprocessedEvent) {
+                $unprocessedEvent->delete();
```

Comment:
I was thinking, in case there are maaaaany lines to delete could this be a bit of a problem? wouldn't be better to just do a "DELETE FROM ... WHERE ..." and maybe also set a LIMIT (I don't know, maybe 10000 records) just to be super safe